components/layout/display_list_builder.rs

```diff
@@ -641,31 +643,35 @@ impl FragmentDisplayListBuilding for Fragment {
                                                          display_list_section: DisplayListSection,
                                                          absolute_bounds: &Rect<Au>,
                                                          clip: &ClippingRegion,
-                                                         gradient: &LinearGradient,
+                                                         gradient: &Gradient,
                                                          style: &ServoComputedValues) {
         let mut clip = clip.clone();
         clip.intersect_rect(absolute_bounds);
 
 
-        let angle = match gradient.angle_or_corner {
-            AngleOrCorner::Angle(angle) => angle.radians(),
-            AngleOrCorner::Corner(horizontal, vertical) => {
-                // This the angle for one of the diagonals of the box. Our angle
-                // will either be this one, this one + PI, or one of the other
-                // two perpendicular angles.
-                let atan = (absolute_bounds.size.height.to_f32_px() /
-                            absolute_bounds.size.width.to_f32_px()).atan();
-                match (horizontal, vertical) {
-                    (HorizontalDirection::Right, VerticalDirection::Bottom)
-                        => f32::consts::PI - atan,
-                    (HorizontalDirection::Left, VerticalDirection::Bottom)
-                        => f32::consts::PI + atan,
-                    (HorizontalDirection::Right, VerticalDirection::Top)
-                        => atan,
-                    (HorizontalDirection::Left, VerticalDirection::Top)
-                        => -atan,
+        let angle = if let GradientKind::Linear(angle_or_corner) = gradient.gradient_kind {
+            match angle_or_corner {
+                AngleOrCorner::Angle(angle) => angle.radians(),
+                AngleOrCorner::Corner(horizontal, vertical) => {
+                    // This the angle for one of the diagonals of the box. Our angle
+                    // will either be this one, this one + PI, or one of the other
+                    // two perpendicular angles.
+                    let atan = (absolute_bounds.size.height.to_f32_px() /
+                                absolute_bounds.size.width.to_f32_px()).atan();
+                    match (horizontal, vertical) {
+                        (HorizontalDirection::Right, VerticalDirection::Bottom)
+                            => f32::consts::PI - atan,
+                        (HorizontalDirection::Left, VerticalDirection::Bottom)
+                            => f32::consts::PI + atan,
+                        (HorizontalDirection::Right, VerticalDirection::Top)
+                            => atan,
+                        (HorizontalDirection::Left, VerticalDirection::Top)
+                            => -atan,
+                    }
                 }
             }
+        } else {
+          0.0 // FIXME(canaltinova): What to do if it's not linear?
```

Comment:
I needed to give it a default value in case of radial gradient. How can we fix that?